featurebyte/worker/task/observation_table.py

```diff
@@ -188,15 +199,94 @@ async def create_observation_table(
             context_id = payload.context_id
             use_case_ids = [payload.use_case_id] if payload.use_case_id else []
 
-        await request_input.materialize(
-            session=db_session,
-            destination=location.table_details,
-            sample_rows=payload.sample_rows,
-            sample_from_timestamp=payload.sample_from_timestamp,
-            sample_to_timestamp=sample_to_timestamp,
-            columns_to_exclude_missing_values=columns_to_exclude_missing_values,
-            missing_data_table_details=missing_data_table_details,
-        )
+        # if use case is specified and observation table does not contain target,
+        # try to compute the target column
+        to_compute_target_id = None
+        target_namespace_id = payload.target_namespace_id
+        if target_namespace_id is None and use_case_ids:
+            # check if use case has target definition
+            use_case = await self.use_case_service.get_document(document_id=use_case_ids[0])
+            if use_case.target_id is not None:
+                to_compute_target_id = use_case.target_id
+
+        if to_compute_target_id is not None:
+            target = await self.target_service.get_document(document_id=to_compute_target_id)
+            target_namespace_id = target.target_namespace_id
+            graph = target.graph
+            node_names = [target.node_name]
+
+            temp_location = copy.deepcopy(location)
+            temp_location.table_details.table_name = f"temp_{location.table_details.table_name}"
```

Comment:
Would be good to use the prefix `__TEMP` to be recognised by the warehouse table cleanup task automatically.